libs/wingc/src/ast.rs

```diff
@@ -118,6 +117,7 @@ pub enum TypeAnnotationKind {
 	Duration,
 	Json,
 	MutJson,
+	Resource,
```

Comment:
Hm - so I couldn't do this before because when I transformed closures like so:
```js
let fn = ((): A => {
  class A {
    // snip
  }
  return new A();
})();
```
... then `A` would not be available outside, so `(): A` was invalid. But since we're now emitting the class in a separate statement, as in...

```
class A {
  // snip
}
let fn = new A();
```
... then I think we can remove this new annotation kind. Let me see what I can do

Reply:
cool